### DISCUSSION TITLE
OOM after instrument

### DISCUSSION THREAD

**Discussion by lmssy** [open-telemetry/opentelemetry-java-instrumentation#3091] [Q&A] [answered]
I meet OOM issue right after instrumenting, here is detail infomation below:

my app use jdk version: 1.8.0_144
opentelemetry-java-instrumentation version 1.2.0

I dump the heap and see below:

![image](https://user-images.githubusercontent.com/17894328/119657539-a0dd6800-be5e-11eb-8c82-10ce5dcec5a0.png)

![image](https://user-images.githubusercontent.com/17894328/119657721-d8e4ab00-be5e-11eb-87b7-bb89b4c552d4.png)

It shows a ConcurrentHashMap which referenced by "io.opentelemetry.javaagent.bootstrap.instrumentation.context.FieldBackedProvider$ContextStore$io$netty$util$concurrent$GenericFutureListener$io$netty$util$concurrent$GenericFutureListener" have 43161 nodes, which cause a memory leak. 

But I can't find the Class "io.opentelemetry.javaagent.bootstrap.instrumentation.context.FieldBackedProvider" in source code.

Is there anyone can explain this? thx in advance!!!

**Comment by trask** [comment, marked as answer]
hey @lmssy! this looks like the same issue as #3027. the fix (#3059) looks like it just barely made it into the latest snapshot build: https://oss.sonatype.org/content/repositories/snapshots/io/opentelemetry/javaagent/opentelemetry-javaagent/1.3.0-SNAPSHOT/, can you give that a try?

**Reply by lmssy** [reply]
hi @trask! thanks for your reply, it works!! 
I pull the latest code from main branch and compile to jar, now there is no OOM.

Very appreciate and wish you a nice day!
